mamba/mamba.py

```diff
@@ -555,7 +554,6 @@ def install(args, parser, command="install"):
 
         pinned_specs_info = ""
         if python_constraint:
-            solver.add_pin(python_constraint)
             pinned_specs_info += f"  - {python_constraint}\n"
```

Comment:
Of course, this isn't actually implemented as a pin now, so I don't know if you would prefer this output to change somehow. It did seem useful to retain something, but I'm happy to tweak. As things stand, it will look like this:

```
Pinned packages:
  - python 3.9.*
```

Reply:
what's wrong about keeping it a `pin`? As far as I can see it's currently added to the regular "install specs"?

Reply:
That was the first thing I tried, naturally, but it didn't seem to work: regardless of how I phrased the looser 'pin spec', it retained the current patch version. For example, I had `python-3.9.4` installed and with a pin of either `python==3.9` or `python==3.9.*` it still kept `3.9.4`. With this regular 'install spec' of `python==3.9.*` instead, it offers `3.9.5`, as it should.

Please let me know if there's some better way. I didn't really want to start digging into the solver side - that seems likely to require beyond the amount of time I can spend on this.

Reply:
(From a very quick at the solver, it seems to me that if `SOLVER_LOCK` is applied to a package, that package will then simply go unchanged, and the rest of its spec is a bit irrelevant.)

Reply:
The `SOLVER_LOCK` is only placed on solvables/packages that are not matching the spec.
So it should work as expected to create a relaxed pin (without the patch).

Reply:
As I said above, that's what I initially tried, e.g.

```python
python_constraint = MatchSpec(f"python=={major_minor_version}").conda_build_form()
```

It doesn't work. This current solution does, though I'm entirely prepared to accept it may not be the ideal approach.